WalletWasabi/BitcoinCore/CoreNode.cs

```diff
@@ -22,134 +26,155 @@ public class CoreNode
+					{
+						whitebind = wb;
+					}
+					if (rpst != null)
 					{
-						var pid = await File.ReadAllTextAsync(pidFile);
-						using var process = Process.GetProcessById(int.Parse(pid));
-						await process.WaitForExitAsync(CancellationToken.None);
+						rpcHost = rpst;
 					}
-					else
+					if (rpts != null && int.TryParse(rpts, out int rpt))
 					{
-						var allProcesses = Process.GetProcesses();
-						var bitcoindProcesses = allProcesses.Where(x => x.ProcessName.Contains("bitcoind"));
-						if (bitcoindProcesses.Count() == 1)
-						{
-							var bitcoind = bitcoindProcesses.First();
-							await bitcoind.WaitForExitAsync(CancellationToken.None);
-						}
+						rpcPort = rpt;
 					}
 				}
-				catch (Exception)
+
+				string authString;
+				bool cookieAuth = rpcCookieFilePath != null;
+				if (cookieAuth)
 				{
+					authString = $"cookiefile={rpcCookieFilePath}";
+				}
+				else
+				{
+					rpcUser ??= Encoders.Hex.EncodeData(RandomUtils.GetBytes(21));
+					rpcPassword ??= Encoders.Hex.EncodeData(RandomUtils.GetBytes(21));
+					authString = $"{rpcUser}:{rpcPassword}";
 				}
-			}
 
-			await IoHelpers.DeleteRecursivelyWithMagicDustAsync(coreNode.DataDir);
-			IoHelpers.EnsureDirectoryExists(coreNode.DataDir);
+				coreNode.P2pEndPoint = whitebind ?? coreNodeParams.P2pEndPointStrategy.EndPoint;
+				rpcHost ??= coreNodeParams.RpcEndPointStrategy.EndPoint.GetHostOrDefault();
+				rpcPort ??= coreNodeParams.RpcEndPointStrategy.EndPoint.GetPortOrDefault();
+				EndPointParser.TryParse($"{rpcHost}:{rpcPort}", coreNode.Network.RPCPort, out EndPoint rpce);
+				coreNode.RpcEndPoint = rpce;
 
-			var pass = Encoders.Hex.EncodeData(RandomUtils.GetBytes(20));
-			var creds = new NetworkCredential(pass, pass);
+				coreNode.RpcClient = new RPCClient($"{authString}", coreNode.RpcEndPoint.ToString(coreNode.Network.DefaultPort), coreNode.Network);
 
-			var portArray = new int[2];
-			var i = 0;
-			while (i < portArray.Length)
-			{
-				var port = RandomUtils.GetUInt32() % 4000;
-				port += 10000;
-				if (portArray.Any(p => p == port))
+				if (coreNodeParams.TryRestart && await coreNode.TryStopAsync().ConfigureAwait(false) && coreNodeParams.TryDeleteDataDir)
 				{
-					continue;
+					await IoHelpers.DeleteRecursivelyWithMagicDustAsync(coreNode.DataDir).ConfigureAwait(false);
 				}
 
-				try
+				IoHelpers.EnsureDirectoryExists(coreNode.DataDir);
+
+				var configPrefix = NetworkTranslator.GetConfigPrefix(coreNode.Network);
+				var desiredConfigLines = new List<string>()
 				{
-					var listener = new TcpListener(IPAddress.Loopback, (int)port);
-					listener.Start();
-					listener.Stop();
-					portArray[i] = (int)port;
-					i++;
-				}
-				catch (SocketException)
+					$"{configPrefix}.server			= 1",
+					$"{configPrefix}.whitebind		= {coreNode.P2pEndPoint.ToString(coreNode.Network.DefaultPort)}",
+					$"{configPrefix}.rpchost		= {coreNode.RpcEndPoint.GetHostOrDefault()}",
```

Comment:
Given we still don't know how the integration will be done, we don't know if we need to `whitebind` or enable `rpc`. However, imo we shouldn't use rpc.